### ISSUE TITLE
Write tests for usages of 'panic_on_migration()'

### ISSUE THREAD

**Issue by bertllll** [MASQ-Project/MASQ-Node-issues#662]
I've realized that my past implementation of DbInitializationConfig::panic_on_migration() has been quite naive. I ignored the fact whether or not the configuration is also tested.

I remember that the introduction of this concept stemmed from my dissatisfaction with our holey knowledge of where a required database migration would be happening in the real life and if we care about watching all places which allow that. My assumption was that the clearest scenario would be to have just a single spot in the whole Node's code to which we can go if we wonder about this kind of question. 

The result of it was that I defined, historically probably four (now it might be slightly adjusted) different configurations as the only ones that were offered to be used for any new implementation of the DB connection initialization. 

Basically there's a singleton allowing database creation and migration. Then at all other places in the Node's code (for Daemon and DumpConfig it's a bit different) are supposed to use `DbInitializationConfig::panic_on_migration()` which lets anybody neither create a new database file nor migrate an existing database. 

Before beginning to write this post, I counted up 7 occurrences of this configuration in the production code. I'm quite sure though that I didn't bother to test these specifics and we felt satisfied with just a visual check. 

Because there aren't many places being affected so, we can add more tests proving that if we try to migrate there we get a reaction typical for this particular configuration: a specific panic.
